frontend/src/features/dsPreview/DatasetPreviewPage.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react'
+import { useSelector } from 'react-redux'
+import useDimensions from 'react-use-dimensions'
+
+import { selectDataset, selectIsDatasetLoading } from '../dataset/state/datasetState'
+import DatasetComponent from '../dsComponents/DatasetComponent'
+import Spinner from '../../chrome/Spinner'
+import ContentBox from '../../chrome/ContentBox'
+import Icon from '../../chrome/Icon'
+import ContentBoxTitle from '../../chrome/ContentBoxTitle'
+import DownloadDatasetButton from '../download/DownloadDatasetButton'
+import RelativeTimestampWithIcon from '../../chrome/RelativeTimestampWithIcon'
+import UsernameWithIcon from '../../chrome/UsernameWithIcon'
+import BodyPreview from '../dsComponents/body/BodyPreview'
+import DatasetHeader from '../dataset/DatasetHeader'
+import NavBar from '../navbar/NavBar'
+import DatasetNavSidebar from '../dataset/DatasetNavSidebar'
+import DeployingScreen from '../deploy/DeployingScreen'
+
+import { selectSessionUserCanEditDataset } from '../dataset/state/datasetState';
+import { QriRef } from '../../qri/ref'
+
+interface DatasetPreviewPageProps {
+  qriRef: QriRef
+}
+
+const DatasetPreviewPage: React.FC<DatasetPreviewPageProps> = ({
+  qriRef
+}) => {
+  const dataset = useSelector(selectDataset)
+  const loading = useSelector(selectIsDatasetLoading)
+  const editable = useSelector(selectSessionUserCanEditDataset)
+
+  const [versionInfoContainer, { height: versionInfoContainerHeight }] = useDimensions();
+  const [expandReadme, setExpandReadme] = useState(false)
+
+  let readmeContainerHeight = versionInfoContainerHeight || 'auto'
+
+  if (expandReadme) {
+    readmeContainerHeight = 'auto'
+  }
+
+  return (
+    <div className='flex flex-col h-full w-full' style={{ backgroundColor: '#f3f4f6'}}>
+      <NavBar />
+      <div className='flex overflow-hidden w-full'>
+        <DatasetNavSidebar qriRef={qriRef} />
+        {loading
+        ? (<div className='w-full h-full p-4 flex justify-center items-center'>
+            <Spinner color='#4FC7F3' />
```

Comment:
another one here. This could be `vivd sky blue`, but I suspect it's already `qri blue` or something to that affect.

Reply:
It is indeed `qriblue-500`, but the `Spinner` component wraps a library that needs a hex color as a prop.  From a few minutes of googling, there's no easy way to translate a custom tailwind color name into a hex color outside of tailwind.  

In theory we can write a lookup utility function that reads from `tailwind.config.js`, but CRA doesn't allow importing this file because it's outside of `/src`.

I'll leave this as-is for now, these spinners may not survive our redesign anyway.